Kokkos + KokkosKernels Promotion to 2.8.00

ndellingwood opened the pull request:
This promotes Kokkos and KokkosKernels to Version 2.8.00.

@trilinos/kokkos @trilinos/kokkos-kernels 

This update includes the following:

### Kokkos
Volta fixes, C++14 support and testing, Capability to add environment variables for all command line arguments, Threads RangePolicy fix for offsets, RemoteMemorySpace support

### KokkosKernels
BLAS TPL + build system fixes (Complex API, fortran mangling), C++14 support and testing, Batched GETRS

----

## Kokkos Changelog

## [2.8.00](https://github.com/kokkos/kokkos/tree/2.8.00) (2019-02-05)
[Full Changelog](https://github.com/kokkos/kokkos/compare/2.7.24...2.8.00)

**Implemented enhancements:**

- Capability, Tests: C++14 support and testing [\#1914](https://github.com/kokkos/kokkos/issues/1914)
- Capability: Add environment variables for all command line arguments [\#1798](https://github.com/kokkos/kokkos/issues/1798)
- Capability: --kokkos-ndevices not working for Slurm [\#1920](https://github.com/kokkos/kokkos/issues/1920)
- View: Undefined behavior when deep copying from and to an empty unmanaged view [\#1967](https://github.com/kokkos/kokkos/issues/1967)
- BuildSystem: nvcc\_wrapper should stop immediately if nvcc is not in PATH [\#1861](https://github.com/kokkos/kokkos/issues/1861)

**Fixed bugs:**

- Cuda: Fix Volta Issues 1 Non-deterministic behavior on Volta, runs fine on Pascal [\#1949](https://github.com/kokkos/kokkos/issues/1949)
- Cuda: Fix Volta Issues 2 CUDA Team Scan gives wrong values on Volta with -G compile flag [\#1942](https://github.com/kokkos/kokkos/issues/1942)
- Cuda: illegal warp sync in parallel\_reduce by functor on Turing 75 [\#1958](https://github.com/kokkos/kokkos/issues/1958)
- Threads: Pthreads backend does not handle RangePolicy with offset correctly [\#1976](https://github.com/kokkos/kokkos/issues/1976)
- Atomics: atomic\_fetch\_oper has no case for Kokkos::complex\<double\> or other 16-byte types [\#1951](https://github.com/kokkos/kokkos/issues/1951)
- MDRangePolicy: Fix zero-length range [\#1948](https://github.com/kokkos/kokkos/issues/1948)
- TeamThreadRange: TeamThreadRange MaxLoc reduce doesnt compile  [\#1909](https://github.com/kokkos/kokkos/issues/1909)

----

## KokkosKernels Changelog

## [2.8.00](https://github.com/kokkos/kokkos-kernels/tree/2.8.00) (2019-02-05)
[Full Changelog](https://github.com/kokkos/kokkos-kernels/compare/2.7.24...2.8.00)

**Implemented enhancements:**

- Capability, Tests: C++14 Support and Testing [\#351](https://github.com/kokkos/kokkos-kernels/issues/351)
- Capability: Batched getrs [\#332](https://github.com/kokkos/kokkos-kernels/issues/332)
- More Kernel Labels for KokkosBlas [\#239](https://github.com/kokkos/kokkos-kernels/issues/239)
- Name all parallel kernels and regions [\#124](https://github.com/kokkos/kokkos-kernels/issues/124)

**Fixed bugs:**

- BLAS TPL: BLAS underscore mangling [\#369](https://github.com/kokkos/kokkos-kernels/issues/369)
- BLAS TPL, Complex: Promotion 2.7.24 broke MV unit tests in Tpetra with complex types [\#360](https://github.com/kokkos/kokkos-kernels/issues/360)
- GEMM: GEMM uses wrong function for computing shared memory allocation size [\#368](https://github.com/kokkos/kokkos-kernels/issues/368)
- BuildSystem: BLAS TPL macro not properly enabled with MKL BLAS [\#347](https://github.com/kokkos/kokkos-kernels/issues/347)
- BuildSystem: make clean - errors [\#353](https://github.com/kokkos/kokkos-kernels/issues/353)
- Compiler Workaround: Internal compiler error in KokkosBatched::Experimental::TeamGemm [\#349](https://github.com/kokkos/kokkos-kernels/issues/349)
- KokkosBlas: Some KokkosBlas kernels assume default execution space [\#14](https://github.com/kokkos/kokkos-kernels/issues/14)

trilinos-autotester commented:
Status Flag 'Pre-Test Inspection' - Auto Inspected - Inspection Is Not Necessary for this Pull Request.

ndellingwood commented:
Cross-referencing the Kokkos promotion 2.8.00 issue (where testing tracked) kokkos/kokkos#1981

@crtrott @srajama1 @ibaned 

Other likely interested parties: @mhoemmen @ZUUL42 @william76

trilinos-autotester commented:
Status Flag 'Pull Request AutoTester' - Testing Jenkins Projects:

<details>
<summary>
  <b>Pull Request Auto Testing STARTING</b> (click to expand)
</summary>

<h2>Build Information</h2>
<h3>Test Name: Trilinos_pullrequest_gcc_4.8.4</h3>
<ul>
  <li>Build Num: 2376</li>
  <li>Status: STARTED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>COMPILER_MODULE</td>
    <td>sems-gcc/4.8.4</td>
  </tr>

  <tr>
    <td>JENKINS_BUILD_TYPE</td>
    <td>Release</td>
  </tr>

  <tr>
    <td>JENKINS_COMM_TYPE</td>
    <td>MPI</td>
  </tr>

  <tr>
    <td>JENKINS_DO_COMPLEX</td>
    <td>OFF</td>
  </tr>

  <tr>
    <td>JENKINS_JOB_TYPE</td>
    <td>Experimental</td>
  </tr>

  <tr>
    <td>MPI_MODULE</td>
    <td>sems-openmpi/1.8.7</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>4329</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_BRANCH</td>
    <td>kokkos-promotion</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>c5c860803c519ab231238dfebe5a78fe8d3b9da7</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>fa4806254b845aaebd000405fc779de94d6644fe</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: Trilinos_pullrequest_intel_17.0.1</h3>
<ul>
  <li>Build Num: 2177</li>
  <li>Status: STARTED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>4329</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_BRANCH</td>
    <td>kokkos-promotion</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>c5c860803c519ab231238dfebe5a78fe8d3b9da7</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>fa4806254b845aaebd000405fc779de94d6644fe</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: Trilinos_pullrequest_gcc_4.9.3_SERIAL</h3>
<ul>
  <li>Build Num: 668</li>
  <li>Status: STARTED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>4329</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_BRANCH</td>
    <td>kokkos-promotion</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>c5c860803c519ab231238dfebe5a78fe8d3b9da7</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>fa4806254b845aaebd000405fc779de94d6644fe</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: Trilinos_pullrequest_gcc_7.2.0</h3>
<ul>
  <li>Build Num: 295</li>
  <li>Status: STARTED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>4329</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_BRANCH</td>
    <td>kokkos-promotion</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>c5c860803c519ab231238dfebe5a78fe8d3b9da7</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>fa4806254b845aaebd000405fc779de94d6644fe</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: Trilinos_pullrequest_cuda_9.2</h3>
<ul>
  <li>Build Num: 80</li>
  <li>Status: STARTED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>JENKINS_JOB_TYPE</td>
    <td>Experimental</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>4329</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_BRANCH</td>
    <td>kokkos-promotion</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>c5c860803c519ab231238dfebe5a78fe8d3b9da7</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>fa4806254b845aaebd000405fc779de94d6644fe</td>
  </tr>
</table>

<h2>Using Repos:</h2>

<dl>
  <dt>Repo: TRILINOS (trilinos/Trilinos)</dt>
  <ul>
    <li>Branch: kokkos-promotion</li>
    <li>SHA: c5c860803c519ab231238dfebe5a78fe8d3b9da7</li>
    <li>Mode: TEST_REPO</li>
  </ul>
</dl>

Pull Request Author: ndellingwood
</details>

trilinos-autotester commented:
Status Flag 'Pull Request AutoTester' - Jenkins Testing: all Jobs PASSED

<details>
<summary>
  <b>Pull Request Auto Testing has PASSED</b> (click to expand)
</summary>

<h2>Build Information</h2>
<h3>Test Name: Trilinos_pullrequest_gcc_4.8.4</h3>
<ul>
  <li>Build Num: 2376</li>
  <li>Status: PASSED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>COMPILER_MODULE</td>
    <td>sems-gcc/4.8.4</td>
  </tr>

  <tr>
    <td>JENKINS_BUILD_TYPE</td>
    <td>Release</td>
  </tr>

  <tr>
    <td>JENKINS_COMM_TYPE</td>
    <td>MPI</td>
  </tr>

  <tr>
    <td>JENKINS_DO_COMPLEX</td>
    <td>OFF</td>
  </tr>

  <tr>
    <td>JENKINS_JOB_TYPE</td>
    <td>Experimental</td>
  </tr>

  <tr>
    <td>MPI_MODULE</td>
    <td>sems-openmpi/1.8.7</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>4329</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_BRANCH</td>
    <td>kokkos-promotion</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>c5c860803c519ab231238dfebe5a78fe8d3b9da7</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>fa4806254b845aaebd000405fc779de94d6644fe</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: Trilinos_pullrequest_intel_17.0.1</h3>
<ul>
  <li>Build Num: 2177</li>
  <li>Status: PASSED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>4329</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_BRANCH</td>
    <td>kokkos-promotion</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>c5c860803c519ab231238dfebe5a78fe8d3b9da7</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>fa4806254b845aaebd000405fc779de94d6644fe</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: Trilinos_pullrequest_gcc_4.9.3_SERIAL</h3>
<ul>
  <li>Build Num: 668</li>
  <li>Status: PASSED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>4329</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_BRANCH</td>
    <td>kokkos-promotion</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>c5c860803c519ab231238dfebe5a78fe8d3b9da7</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>fa4806254b845aaebd000405fc779de94d6644fe</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: Trilinos_pullrequest_gcc_7.2.0</h3>
<ul>
  <li>Build Num: 295</li>
  <li>Status: PASSED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>4329</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_BRANCH</td>
    <td>kokkos-promotion</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>c5c860803c519ab231238dfebe5a78fe8d3b9da7</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>fa4806254b845aaebd000405fc779de94d6644fe</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: Trilinos_pullrequest_cuda_9.2</h3>
<ul>
  <li>Build Num: 80</li>
  <li>Status: PASSED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>JENKINS_JOB_TYPE</td>
    <td>Experimental</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>4329</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_BRANCH</td>
    <td>kokkos-promotion</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>c5c860803c519ab231238dfebe5a78fe8d3b9da7</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>fa4806254b845aaebd000405fc779de94d6644fe</td>
  </tr>
</table>
</details>


<br>[CDash Test Results for PR# 4329.](https://testing-vm.sandia.gov/cdash/index.php?project=Trilinos&filtercombine=and&filtercombine=and&filtercombine=and&filtercombine=and&filtercombine=and&filtercombine=and&filtercount=2&showfilters=1&filtercombine=and&field1=buildname&compare1=63&value1=PR-4329-test&field2=buildstarttime&compare2=84&value2=NOW
)<br>

trilinos-autotester commented:
Status Flag 'Pre-Merge Inspection' - SUCCESS: The last commit to this Pull Request has been INSPECTED AND APPROVED by [ srajama1 ]!

trilinos-autotester commented:
Status Flag 'Pull Request AutoTester' - AutoMerge IS ENABLED, but the Label AT: AUTOMERGE is not set.  Either set Label AT: AUTOMERGE or manually merge the PR...

ndellingwood commented:
Testing results from kokkos/kokkos#1981

# Testing 1

* Testing with scripts from kokkos-kernels/scripts/trilinos_integration

## White Cuda Kepler Nodes
Trilinos pristine: develop branch          SHA trilinos/Trilinos@80e2cb3256ef6f7b3046343112ee44444069c4b0
Trilinos updated:  kokkos-promotion branch SHA trilinos/Trilinos@80e2cb3256ef6f7b3046343112ee44444069c4b0 (before snapshot)
                   kokkos develop branch   SHA kokkos/kokkos@534b7f32597b885097b91873edf74be02079a5dc
                   kokkos-kernels develop  SHA kokkos/kokkos-kernels@52a6ba2e4fb5e1a66b87b8208903e4bbc9cdc036

### Summary
Compilation - No new build failures
Tests - One new failure
  `Stokhos_TpetraCrsMatrixMPVectorUnitTest_Serial_MPI_4`

  This test failure was fixed with PR kokkos/kokkos-kernels#381 pattern matching existing fix in Trilinos that tests this case.

  Results of a rerun with that updated:
```
bash-4.2$ ctest -R Stokhos_TpetraCrsMatrixMPVectorUnitTest_Serial_MPI_4
Test project /ascldap/users/ndellin/IntegrationTests-White/cuda-kepler/build_updated
    Start 1683: Stokhos_TpetraCrsMatrixMPVectorUnitTest_Serial_MPI_4
1/1 Test #1683: Stokhos_TpetraCrsMatrixMPVectorUnitTest_Serial_MPI_4 ...   Passed    4.80 sec
```

----

## White OpenMP
Trilinos pristine: develop branch          SHA trilinos/Trilinos@80e2cb3256ef6f7b3046343112ee44444069c4b0
Trilinos updated:  kokkos-promotion branch SHA trilinos/Trilinos@80e2cb3256ef6f7b3046343112ee44444069c4b0 (before snapshot)
                   kokkos develop branch   SHA kokkos/kokkos@534b7f32597b885097b91873edf74be02079a5dc
                   kokkos-kernels develop  SHA kokkos/kokkos-kernels@52a6ba2e4fb5e1a66b87b8208903e4bbc9cdc036

### Summary
Compilation - No new build failures
Tests - No new test failures

----

## White Cuda + Complex_Double Kepler Nodes
Trilinos pristine: develop branch          SHA trilinos/Trilinos@ac4762eb8464b8f58c53cc991be3fcc70563f9da
Trilinos updated:  kokkos-promotion branch SHA trilinos/Trilinos@80e2cb3256ef6f7b3046343112ee44444069c4b0 (before snapshot)
                   kokkos develop branch   SHA kokkos/kokkos@534b7f32597b885097b91873edf74be02079a5dc
                   kokkos-kernels develop  SHA kokkos/kokkos-kernels@67af7daac3690a94697280bb540f103f33b6a17a

### Summary
Compilation - No new build failures
Tests - No new test failures

----

## Blake Serial
Trilinos pristine: develop branch          SHA trilinos/Trilinos@80e2cb3256ef6f7b3046343112ee44444069c4b0
Trilinos updated:  kokkos-promotion branch SHA trilinos/Trilinos@80e2cb3256ef6f7b3046343112ee44444069c4b0 (before snapshot)
                   kokkos develop branch   SHA kokkos/kokkos@534b7f32597b885097b91873edf74be02079a5dc
                   kokkos-kernels develop  SHA kokkos/kokkos-kernels@52a6ba2e4fb5e1a66b87b8208903e4bbc9cdc036

### Summary
Compilation - No new build failures
Tests - One new failure
  `Stokhos_TpetraCrsMatrixMPVectorUnitTest_Serial_MPI_4`

  This test failure was fixed with PR kokkos/kokkos-kernels#381 pattern matching existing fix in Trilinos that tests this case.

----

## Blake Pthreads
Trilinos pristine: develop branch          SHA trilinos/Trilinos@80e2cb3256ef6f7b3046343112ee44444069c4b0
Trilinos updated:  kokkos-promotion branch SHA trilinos/Trilinos@80e2cb3256ef6f7b3046343112ee44444069c4b0 (before snapshot)
                   kokkos develop branch   SHA kokkos/kokkos@534b7f32597b885097b91873edf74be02079a5dc
                   kokkos-kernels develop  SHA kokkos/kokkos-kernels@52a6ba2e4fb5e1a66b87b8208903e4bbc9cdc036

### Summary
Compilation - No new build failures
Tests - One new failure
  `Stokhos_TpetraCrsMatrixMPVectorUnitTest_Serial_MPI_4`

  This test failure was fixed with PR kokkos/kokkos-kernels#381 pattern matching existing fix in Trilinos that tests this case.

----

# Testing 2

* Various tests using ATDM configurations and packages highly impacted by Kokkos modifications

## White cuda-9.2-opt-Kepler37

### Build 1: Updated Trilinos, deprecated code disabled

Trilinos updated:  kokkos-promotion branch SHA trilinos/Trilinos@80e2cb3256ef6f7b3046343112ee44444069c4b0 (before snapshot)
                   kokkos develop branch   SHA kokkos/kokkos@534b7f32597b885097b91873edf74be02079a5dc
                   kokkos-kernels develop  SHA kokkos/kokkos-kernels@67af7daac3690a94697280bb540f103f33b6a17a

#### Reproducer instructions

```
module purge
source ${TRILINOS_DIR}/cmake/std/atdm/load-env.sh cuda-9.2-opt-Kepler37
module swap cmake/3.9.6 cmake/3.12.3

cmake \
 -GNinja \
 -DTrilinos_CONFIGURE_OPTIONS_FILE:STRING=cmake/std/atdm/ATDMDevEnv.cmake \
 -DTrilinos_ENABLE_TESTS=ON \
  -DKOKKOS_ENABLE_DEPRECATED_CODE:BOOL=OFF \
  -DKokkos_SOURCE_DIR_OVERRIDE:STRING=kokkos \
  -DKokkosKernels_SOURCE_DIR_OVERRIDE:STRING=kokkos-kernels \
  -DTrilinos_ENABLE_MueLu=ON \
    -DMueLu_ENABLE_Experimental:BOOL=ON \
    -DMueLu_ENABLE_Kokkos_Refactor:BOOL=ON \
    -DMueLu_ENABLE_Epetra:BOOL=OFF \
    -DXpetra_ENABLE_Experimental:BOOL=ON \
    -DMueLu_ENABLE_TESTS:BOOL=OFF \
    -DMueLu_ENABLE_EXAMPLES:BOOL=OFF \
  -DTrilinos_ENABLE_Tpetra=ON \
  -DTrilinos_ENABLE_Sacado=ON \
  -DTrilinos_ENABLE_Intrepid2=ON \
  -DTrilinos_ENABLE_Panzer=ON \
$TRILINOS_DIR
```

#### Summary

Compilation - No build failures
Tests - 4 new failures, however they passed when rerun with ctest (no -j)

```
        867 - PanzerAdaptersSTK_STK_VolumeSideResponse_MPI_2 (Timeout) #NEW - passed with ctest with no -j
        868 - PanzerAdaptersSTK_ip_coordinates_MPI_2 (Timeout) #NEW - passed with ctest with no -j
        874 - PanzerAdaptersSTK_tDomainInterface_MPI_2 (Timeout) #NEW - passed with ctest with no -j
        875 - PanzerAdaptersSTK_node_normals_MPI_2 (Timeout) #NEW - passed with ctest with no -j
```

Several pre-existing Panzer failures when deprecated code disabled (occurring in builds with Trilnios VOTD develop branch). Failures indicate view(s) constructed with more arguments than dynamic rank.

----

## White cuda-9.2-debug-Kepler37

### Build 2: Updated Trilinos, deprecated code disabled, debugging on

Trilinos updated:  kokkos-promotion branch SHA trilinos/Trilinos@80e2cb3256ef6f7b3046343112ee44444069c4b0 (before snapshot)
                   kokkos develop branch   SHA kokkos/kokkos@534b7f32597b885097b91873edf74be02079a5dc
                   kokkos-kernels develop  SHA kokkos/kokkos-kernels@67af7daac3690a94697280bb540f103f33b6a17a

#### Reproducer instructions

```
module purge
source $TRILINOS_DIR/cmake/std/atdm/load-env.sh cuda-9.2-debug-Kepler37
module swap cmake/3.9.6 cmake/3.12.3

cmake \
 -GNinja \
 -DTrilinos_CONFIGURE_OPTIONS_FILE:STRING=cmake/std/atdm/ATDMDevEnv.cmake \
 -DTrilinos_ENABLE_TESTS=ON \
  -DTrilinos_ENABLE_Tpetra=ON \
  -DTrilinos_ENABLE_Sacado=ON \
  -DTrilinos_ENABLE_Stokhos=ON \
  -DKOKKOS_ENABLE_DEPRECATED_CODE=OFF \
  -DKokkos_SOURCE_DIR_OVERRIDE:STRING=kokkos \
  -DKokkosKernels_SOURCE_DIR_OVERRIDE:STRING=kokkos-kernels \
$TRILINOS_DIR
```

#### Summary

Compilation - No build failures
Tests - No new failures

Several existing Panzer failures when deprecated code disabled (View constructed with more arguments than dynamic rank)

Same as above

----

## Waterman cuda-9.2-opt (Volta)

### Build: Updated Trilinos, deprecated code disabled, debugging on

Trilinos updated:  kokkos-promotion branch SHA trilinos/Trilinos@80e2cb3256ef6f7b3046343112ee44444069c4b0 (before snapshot)
                   kokkos develop branch   SHA kokkos/kokkos@534b7f32597b885097b91873edf74be02079a5dc
                   kokkos-kernels develop  SHA kokkos/kokkos-kernels@67af7daac3690a94697280bb540f103f33b6a17a

#### Reproducer instructions

```
module purge
source ${TRILINOS_DIR}/cmake/std/atdm/load-env.sh cuda-9.2-opt
module swap cmake/3.6.2 cmake/3.12.3

PACKAGE1=Tpetra
PACKAGE2=Sacado
PACKAGE3=Stokhos
PACKAGE4=MueLu
PACKAGE5=Intrepid2
PACKAGE6=Ifpack2
PACKAGE7=Panzer
PACKAGE8=Phalanx
PACKAGE9=Stratimikos
PACKAGE10=Belos

cmake \
 -GNinja \
 -DTrilinos_CONFIGURE_OPTIONS_FILE:STRING=cmake/std/atdm/ATDMDevEnv.cmake \
 -DTrilinos_ENABLE_TESTS=ON \
  -DTrilinos_ENABLE_${PACKAGE1}=ON \
  -DTrilinos_ENABLE_${PACKAGE2}=ON \
  -DTrilinos_ENABLE_${PACKAGE3}=ON \
  -DTrilinos_ENABLE_${PACKAGE4}=ON \
  -DTrilinos_ENABLE_${PACKAGE5}=ON \
  -DTrilinos_ENABLE_${PACKAGE6}=ON \
  -DTrilinos_ENABLE_${PACKAGE7}=ON \
  -DTrilinos_ENABLE_${PACKAGE8}=ON \
  -DTrilinos_ENABLE_${PACKAGE9}=ON \
  -DTrilinos_ENABLE_${PACKAGE10}=ON \
  -DKOKKOS_ENABLE_DEPRECATED_CODE=OFF \
  -DKokkos_SOURCE_DIR_OVERRIDE:STRING=kokkos \
  -DKokkosKernels_SOURCE_DIR_OVERRIDE:STRING=kokkos-kernels \
$TRILINOS_DIR
```

#### Summary

Compilation - Build failures in Ifpack2 (deprecated code disabled) fixed in kokkos-promotion branch
Tests - No new failures

Several existing Panzer failures when deprecated code disabled (View constructed with more arguments than dynamic rank)

Same as above

----

## Blake Pthreads Retesting

With PR kokkos/kokkos#1980

Trilinos pristine: develop branch SHA trilinos/Trilinos@967ff0ab8cd94503c8f52fafcd991328f628f912
Trilinos updated: kokkos-promotion branch SHA trilinos/Trilinos@05a91ffca63eaf12de731c83606a76e20f0d7b82 (merge commit trilinos/Trilinos@b858cf87ed9c0d05ccdeb5f4840dd2e8999b6120)
kokkos issue-1976b branch SHA kokkos/kokkos@f782e0ab09b8d4f5ab797fdbacc922f6aa6cda9d
kokkos-kernels develop SHA kokkos/kokkos-kernels@67af7daac3690a94697280bb540f103f33b6a17a

#### Summary
No new build failures
No new test failures

Existing test failures for reference:

```
MueLu_ImportPerformance_Tpetra_MPI_4 (Failed)
MueLu_Q2Q1-Tpetra_MPI_1 (Failed)
ROL_adapters_tpetra_test_sol_TpetraSimulatedConstraintInterfaceCVaR_MPI_4 (Failed)
Stratimikos_test_single_belos_thyra_solver_driver_nos1_nrhs8_MPI_1 (Failed)
```

ndellingwood commented:
PR testing successful, merging!
